vulture_os/services/frontend/models.py

```diff
@@ -527,10 +533,90 @@ class Frontend(models.Model):
+        verbose_name=_("Size of the queue in nb of message"),
+        validators=[MinValueValidator(100)]
+    )
+    dequeue_batch_size = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_("Size of the batch to dequeue"),
+        verbose_name=_("Size of the batch to dequeue"),
+        validators=[MinValueValidator(1)]
+    )
     nb_workers = models.PositiveIntegerField(
         default=8,
         help_text=_("Maximum number of workers for rsyslog ruleset"),
-        verbose_name=_("Maximum parser workers")
+        verbose_name=_("Maximum parser workers"),
+        validators=[MinValueValidator(1)]
+    )
+    new_worker_minimum_messages = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_("Number of messages in queue to start a new worker thread"),
+        verbose_name=_("Minimum messages to start a new worker"),
+        validators=[MinValueValidator(1)]
+    )
+    shutdown_timeout = models.PositiveIntegerField(
+        default=5000,
+        blank=True,
+        null=True,
+        help_text=_("Time to wait for the queue to finish processing entries (in ms)"),
+        verbose_name=_("Queue timeout shutdown (ms)"),
+        validators=[MinValueValidator(1)]
+    )
+    enable_disk_assist = models.BooleanField(
+        default=False,
+        help_text=_("Enable disk-assisted queue on failure"),
+        verbose_name=_("Enable disk queue on failure")
+    )
+    high_watermark = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_("Write in DA queue after nb messages"),
+        verbose_name=_("High watermark target"),
+        validators=[MinValueValidator(100)]
+    )
+    low_watermark = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_("Write in DA queue until nb messages"),
+        verbose_name=_("Low watermark target"),
+        validators=[MinValueValidator(100)]
+    )
+    max_file_size = models.PositiveIntegerField(
+        default=256,
+        blank=True,
+        null=True,
+        help_text=_("Set the max value of the queue in MB"),
+        verbose_name=_("Max file size of the queue in MB"),
+        validators=[MinValueValidator(1)]
+    )
+    max_disk_space = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_("Limit the maximum disk space used by the queue in MB"),
+        verbose_name=_("Max disk space used by the queue in MB"),
+        validators=[MinValueValidator(1)]
+    )
+    spool_directory = models.TextField(
+        default="/var/tmp",
+        help_text=_("Set a writable folder to store DA queue"),
+        verbose_name=_("DA queue folder")
+    )
+    save_on_shutdown = models.BooleanField(
+        default=True,
+        help_text=_("Prevent message loss by writing on disk during a shutdown"),
+        verbose_name=_("Save in-memory queue on shutdown")
```

Comment:
Could we instead maybe create a separate Model for the Rsyslog queue/action parameters, and then use a OneToOneField to map to it?